config/dynamic.c

```diff
@@ -16,11 +16,467 @@
+				conn_fd, msg_buff, msg_len, dy_conf);
+			if (ret < 0)
+				return -1;
+
+			break;
+		}
+	}
+
+	/* The connection with the client is closed. */
+	if (recv_size == 0) {
+		RTE_LOG(NOTICE, GATEKEEPER,
+			"dyn_cfg: Client %s disconnected!\n",
+			client_ip);
+		return 1;
+	}
+
+	return 0;
+}
+
+static void
+handle_client(int server_socket_fd, struct dynamic_config *dy_conf)
+{
+	int ret;
+	int conn_fd;
+	socklen_t len;
+	struct sockaddr_storage client_addr;
+	char client_ip[INET6_ADDRSTRLEN];
+	int client_port;
+	char *msg_buff = rte_calloc(
+			"msg_buff", LUA_MSG_MAX_LEN + 1, sizeof(char), 0);
+	if (msg_buff == NULL) {
+		RTE_LOG(ERR, MALLOC,
+			"dyn_cfg: Failed to allocate message buffer!\n");
+		return;
+	}
+
+	len = sizeof(client_addr);
+	conn_fd = accept(server_socket_fd,
+		(struct sockaddr *)&client_addr, &len);
+        if (conn_fd < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to accept a new connection!\n");
+		return;
+	}
+
+	/*
+	 * All the clients will be accepted as IPv6 clients.
+	 * Note that, if the client is an IPv4 client,
+	 * the address will be shown as an IPv4 Mapped IPv6 address.
+	 */
+	if (likely(client_addr.ss_family == AF_INET6)) {
+    		struct sockaddr_in6 *addr =
+			(struct sockaddr_in6 *)&client_addr;
+    		client_port = ntohs(addr->sin6_port);
+    		inet_ntop(AF_INET6,
+			&addr->sin6_addr, client_ip, sizeof(client_ip));
+	} else {
+		RTE_LOG(WARNING, GATEKEEPER,
+			"Unexpected condition: unknown client type %d at %s",
+			client_addr.ss_family, __func__);
+		return;
+	}
+
+	RTE_LOG(NOTICE, GATEKEEPER,
+		"dyn_cfg: Accepting a connection from client [IP = %s, Port = %d].\n",
+		client_ip, client_port);
+
+	/*
+	 * XXX The request must be received under a specified timeout,
+	 * or the request is aborted. This code needs further test.
+	 */
+	setsockopt(conn_fd, SOL_SOCKET, SO_RCVTIMEO,
+		(const char*)&dy_conf->rcv_time_out,sizeof(struct timeval));
+
+	setsockopt(conn_fd, SOL_SOCKET,
+		SO_RCVBUF, &msg_buff, sizeof(msg_buff));
+
+	while (1) {
+		ret = process_single_cmd(
+			conn_fd, msg_buff, client_ip, dy_conf);
+		if (ret != 0)
+			break;
+	}
+
+	rte_free(msg_buff);
+
+	ret = close(conn_fd);
+	if (ret < 0)
+		perror("Failed to close the connection socket");
+}
+
+static void
+cleanup_dy(struct dynamic_config *dy_conf)
+{
+	int ret;
+
+	if (dy_conf->sock_fd >= 0) {
+		ret = close(dy_conf->sock_fd);
+		if (ret < 0)
+			perror("Failed to close the server socket");
+
+		dy_conf->sock_fd = -1;
+	}
+
+	if (dy_conf->lua_state != NULL) {
+		lua_close(dy_conf->lua_state);
+		dy_conf->lua_state = NULL;
+	}
+}
+
+static int
+dyn_cfg_proc(void *arg)
+{
+	int ret = 0;
+	int ip_reusable = 1;
+	struct sockaddr_in6 server_addr;
+	uint32_t lcore = rte_lcore_id();
+	struct dynamic_config *dy_conf = arg;
+
+	RTE_LOG(NOTICE, GATEKEEPER,
+		"dyn_cfg: the Dynamic Config block is running at lcore = %u\n", lcore);
+
+	/* Init the server socket. */
+    	dy_conf->sock_fd = socket(AF_INET6, SOCK_STREAM, 0);
```

Comment:
I think that instead of using an IP socket for the communication with the dynamic configuration block, we should use a Unix socket. This simplifies production management of Gatekeeper and our code since it doesn't require some sort of authentication.

Just to avoid ambiguity, you need to replace `AF_INET6` with `AF_UNIX` and adjust the code accordingly.